src/freertos_drivers/esp_idf/EspIdfWiFi.hxx

```diff
@@ -0,0 +1,626 @@
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ * ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ * LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ * CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ * SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ * INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ * CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ * ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ * @file EspIdfWiFi.hxx
+ *
+ * Interface for interacting with WiFi services.
+ *
+ * @author Balazs Racz, extended by Stuart Baker
+ * @date 27 Aug 2024, extended starting Jun 17 2025
+ */
+
+#ifndef _FREERTOS_DRIVERS_ESP_IDF_ESPIDFWIFI_HXX_
+#define _FREERTOS_DRIVERS_ESP_IDF_ESPIDFWIFI_HXX_
+
+#include <esp_netif.h>
+#include <nvs_flash.h>
+
+#include "executor/Service.hxx"
+#include "executor/StateFlow.hxx"
+#include "freertos_drivers/common/WiFiInterface.hxx"
+
+#include <sys/socket.h>
+
+#include <list>
+
+#if OPENMRN_HAVE_BSD_SOCKETS_IPV6
+#if LWIP_IPV6
+#define ESP_IDF_WIFI_IPV6
+#endif
+#endif
+
+/// Object for managing WiFi and network interfaces. The following sequence
+/// must be called within or after app_main() in the following order:
+///   -# init()
+///       - performs initialize before starting WiFi
+///   -# setup_ap()
+///       - option, can be used to override the stored AP profile in bootloader
+///   -# start()
+///       - starts the WiFi based on config (AP, STA, or AP + STA mode)
+///
+/// By default, all user configuration is volatile, and will be forgotten upon
+/// reboot. This is useful for a bootloader that just needs to connect to the
+/// "last" known STA and broadcast the "default" AP. Bootloader example:
+///
+/// @code
+/// EspIdfWifi wifi;
+///
+/// void app_main(void)
+/// {
+///     ESP_ERROR_CHECK(nvs_flash_init());
+///     wifiMgr.init();
+///     wifiMgr.setup_ap(wifiMgr.get_hostname().c_str(),
+///         wifiMgr.DEFAULT_PASSWORD, wifiMgr.SEC_WPA2);
+///     wifiMgr.start(WlanRole::AP_STA);
+/// }
+/// @endcode
+///
+/// In order to support non-volatile configuration, use the specialized version
+/// of this object EspIdfWiFiWithConfigUpdateListener instead. In this version,
+/// whenever there is an update to the user configuration, an update flow is
+/// triggered which will commit the user config to both NVS and a companion
+/// MemorySpace.
+///
+class EspIdfWiFi : public WiFiInterface, public StateFlowBase, protected OSMutex
+{
+public:
+    /// Default wifi password.
+    static constexpr char DEFAULT_PASSWORD[] = "123456789";
+
+    /// Constructor.
+    /// @param service service to bind this object to.
```

Comment:
From @balazsracz: Please describe whether we will block this executor or not (e.g. whether it is fine to provide the shared global executor for the stack here).